### ISSUE TITLE
It's unclear in the paper why this regularizer is used

### ISSUE THREAD

**Issue by anandnimkar** [JohannesAck/tf2multiagentrl#6]
https://github.com/JohannesAck/tf2multiagentrl/blob/8b63174b5d314cbf29a7ff2414d5999a0224c03d/tf2marl/agents/maddpg.py#L205

Please can you provide some references or intuition on why this line is in the code? I have found it difficult to find the source of this in the DDPG, MADDPG, and MATD3 papers.

**Comment by anandnimkar** [comment]
I see that it is applied to the policy loss at 1e-3 in the next line:
https://github.com/JohannesAck/tf2multiagentrl/blob/8b63174b5d314cbf29a7ff2414d5999a0224c03d/tf2marl/agents/maddpg.py#L206

**Comment by JohannesAck** [comment]
Hi,
Thank you for your interest!
The main reason the line is here is that it was also used in the original MADDPG code here: https://github.com/openai/maddpg/blob/3ceefa0ada3ff31d633dd0bde8ff95213ce99be3/maddpg/trainer/maddpg.py#L46

As for an interpretation, it simply acts as a regularizer of the policy logits before the gumbel-softmax.
